libsupport/src/ArrowInterchange.cpp

```diff
@@ -0,0 +1,59 @@
+#include "katana/ArrowInterchange.h"
+
+#include <numeric>
+
+#include "katana/Random.h"
+
+namespace {
+std::shared_ptr<arrow::ChunkedArray>
+IndexedTake(
+    const std::shared_ptr<arrow::ChunkedArray>& original,
+    std::shared_ptr<arrow::Array> indices) {
+  // Use Take to select those indices
+  arrow::Result<arrow::Datum> take_result =
+      arrow::compute::Take(original, indices);
+  KATANA_LOG_ASSERT(take_result.ok());
+  arrow::Datum take = std::move(take_result.ValueOrDie());
+  std::shared_ptr<arrow::ChunkedArray> chunked = take.chunked_array();
+  KATANA_LOG_ASSERT(chunked->num_chunks() == 1);
+  return chunked;
+}
+
+}  // anonymous namespace
+
+std::shared_ptr<arrow::Array>
+katana::Unchunk(const std::shared_ptr<arrow::ChunkedArray>& original) {
+  int64_t length = original->length();
+  // Build indices array, reusable across properties
+  arrow::CTypeTraits<int64_t>::BuilderType builder;
+  auto res = builder.Reserve(length);
+  KATANA_LOG_ASSERT(res.ok());
+  for (int64_t i = 0; i < length; ++i) {
+    builder.UnsafeAppend(i);
+  }
+  std::shared_ptr<arrow::Array> indices;
+  res = builder.Finish(&indices);
+  KATANA_LOG_ASSERT(res.ok());
+  auto chunked = IndexedTake(original, indices);
+  return chunked->chunk(0);
+}
```

Comment:
I'm imagining cases where it would be desirable to Unchunk multiple arrays, in which case caching the indices would be valuable. No change needed now

Reply:
Sure that sounds cool.  When you find an application for that, please structure this code to make it fit.